libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapMatcherResult.kt

```diff
@@ -0,0 +1,70 @@
+package com.mapbox.navigation.core.trip.session
+
+import android.location.Location
+
+/**
+ * Provides information about the status of the enhanced location updates generated
+ * by the map matching engine of the Navigation SDK.
+ *
+ * @param enhancedLocation the best possible location update, snapped to the route or map matched to the road if possible.
+ * Equal to data from [LocationObserver.onEnhancedLocationChanged].
+ * @param keyPoints a list (can be empty) of predicted location points leading up to the target update.
+ * The last point on the list (if not empty) is always equal to [enhancedLocation].
+ * Equal to data from [LocationObserver.onEnhancedLocationChanged].
+ * @param isOffRoad whether the SDK thinks that the user is off road, based on the [offRoadProbability].
+ * @param offRoadProbability probability that the user is off road.
+ * @param isTeleport returns true if map matcher changed its opinion about most probable path on last update.
+ * In practice it means we don't need to animate puck movement from previous to current location
+ * and just do an immediate transition instead.
+ * @param roadEdgeMatchProbability when map matcher snaps to a road, this is the confidence in the chosen edge from all nearest edges.
+ */
+class MapMatcherResult internal constructor(
```

Comment:
Could we add a `BuilderTest`?

Reply:
This class is not a builder.

Reply:
Would be great to test `equals` and `hashCode`. Maybe we can extract `equalsTest` and `hashCodeTest` out from `BuilderTest`? Not a blocker for this PR but something to think about 🤔